### PR TITLE
Port : Truncate component property value

### PR DESCRIPTION
**Pull request by sahibamittal**
### Description

Truncates ComponentProperty values.

This is both to prevent the database's length constraint from being violated, and to avoid storing unnecessary long values. Some tools may misuse properties, and in those cases having the complete value is usually not of interest to users.

### Addressed Issue

Backport change https://github.com/DependencyTrack/hyades/issues/1190

### Checklist

- [x] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- [x] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [ ] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have updated the [migration changelog](https://github.com/DependencyTrack/hyades-apiserver/tree/main/src/main/resources/migration) accordingly
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/hyades/tree/main/docs) accordingly

### CODE REVIEW

**Comment by nscuro** [comment]
The limit should also apply to the database schema:

https://github.com/DependencyTrack/hyades-apiserver/blob/e0f083e3d2ffef0c30cba22fba6b36e4148f2aa5/src/main/resources/migration/changelog-v5.5.0.xml#L115-L134

On the same note, could you also add the limits for the other fields so they're in sync with what's defined on the Java model?

**Comment by sahibamittal** [comment]
> The limit should also apply to the database schema:
> 
> https://github.com/DependencyTrack/hyades-apiserver/blob/e0f083e3d2ffef0c30cba22fba6b36e4148f2aa5/src/main/resources/migration/changelog-v5.5.0.xml#L115-L134
> 
> On the same note, could you also add the limits for the other fields so they're in sync with what's defined on the Java model?

Sure, though fields other than `propertyValue` don't have length constraints in java model, do you want me add them like in upstream?

**Comment by nscuro** [comment]
Argh, apologies I was looking at the upstream code indeed. I think it makes sense to add them for consistency, WDYT?

**Review by nscuro — DISMISSED** [review]

**Comment by nscuro** [comment]
Seems the new SARIF test is a bit flaky:

```
 Error:  Failures: 
Error:    FindingResourceTest.getSARIFFindingsByProjectTest:695 JSON documents are different:
Different value found in node "runs[0].results[1].level", expected: <"note"> but was: <"error">.
Different value found in node "runs[0].results[1].message.text", expected: <"A description-with-hyphens-(and parentheses)"> but was: <"Yet another description but with surrounding whitespaces">.
Different value found in node "runs[0].results[1].properties.cweId", expected: <"23"> but was: <"46">.
Different value found in node "runs[0].results[1].properties.recommendation", expected: <"Recommendation with whitespaces"> but was: <"">.
Different value found in node "runs[0].results[1].properties.severityRank", expected: <"3"> but was: <"1">.
Different value found in node "runs[0].results[1].ruleId", expected: <"Vuln-3"> but was: <"Vuln-2">.
Different value found in node "runs[0].results[2].level", expected: <"error"> but was: <"note">.
Different value found in node "runs[0].results[2].message.text", expected: <"Yet another description but with surrounding whitespaces"> but was: <"A description-with-hyphens-(and parentheses)">.
Different value found in node "runs[0].results[2].properties.cweId", expected: <"46"> but was: <"23">.
Different value found in node "runs[0].results[2].properties.recommendation", expected: <""> but was: <"Recommendation with whitespaces">.
Different value found in node "runs[0].results[2].properties.severityRank", expected: <"1"> but was: <"3">.
Different value found in node "runs[0].results[2].ruleId", expected: <"Vuln-2"> but was: <"Vuln-3">.
```

Do you see any way we can stabilize it?

File: src/test/java/org/dependencytrack/resources/v1/FindingResourceTest.java [hunk 1]
```diff
@@ -688,174 +690,135 @@ public void getSARIFFindingsByProjectTest() {
         Assert.assertEquals(200, response.getStatus(), 0);
         Assert.assertEquals(MEDIA_TYPE_SARIF_JSON, response.getHeaderString(HttpHeaders.CONTENT_TYPE));
         final String jsonResponse = getPlainTextBody(response);
+        JSONArray resultArray = new JSONObject(jsonResponse).getJSONArray("runs").getJSONObject(0).getJSONArray("results");
 
         assertThatJson(jsonResponse)
                 .withMatcher("version", equalTo(new About().getVersion()))
-                .withMatcher("fullName", equalTo("OWASP Dependency-Track - " + new About().getVersion()))
```
Comment (nscuro) [inline src/test/java/org/dependencytrack/resources/v1/FindingResourceTest.java:694]: You can achieve the same outcome without the need to parse intermediary objects with:

https://github.com/DependencyTrack/hyades-apiserver/blob/a48b7973514b9c07bf6a84de24651fac8e395dce/src/test/java/org/dependencytrack/model/mapping/PolicyProtoMapperTest.java#L79

**Review by nscuro — APPROVED** [review]

**Pull request merged**